juriscraper/opinions/united_states/state/tex.py

```diff
@@ -14,298 +14,104 @@
-                    defendant = html_tree.xpath(
-                        "//text()[contains(., 'v.:')]/ancestor::div[@class='span2']/following-sibling::div/text()"
-                    )[0]
-                except IndexError:
-                    logger.warning(f"No title or defendant found for {url}")
-
-                if defendant.strip():
-                    # If there's a defendant
-                    return titlecase(f"{plaintiff} v. {defendant}")
-                else:
-                    return titlecase(plaintiff)
-
-        seed_urls = []
-        for html_tree in self.html:
-            page_records_count = self._get_opinion_count(html_tree)
-            for record in range(page_records_count):
-                path = "id('ctl00_ContentPlaceHolder1_grdDocuments_ctl00__{n}')/td[5]//@href".format(
-                    n=record
-                )
-                seed_urls.append(html_tree.xpath(path)[0])
-        if seed_urls:
-            return DeferringList(seed=seed_urls, fetcher=fetcher)
-        else:
-            return []
-
-    def _get_case_dates(self):
-        for html_tree in self.html:
-            page_records_count = self._get_opinion_count(html_tree)
-            for record in range(page_records_count):
-                path = "id('ctl00_ContentPlaceHolder1_grdDocuments_ctl00__{n}')/td[2]/text()".format(
-                    n=record
-                )
-                yield datetime.strptime(html_tree.xpath(path)[0], "%m/%d/%Y")
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * self.records_nr
-
-    def _get_download_urls(self):
-        for html_tree in self.html:
-            page_records_count = self._get_opinion_count(html_tree)
-            for record in range(page_records_count):
-                path = "id('ctl00_ContentPlaceHolder1_grdDocuments_ctl00__{n}')/td[4]//@href".format(
-                    n=record
-                )
-                yield html_tree.xpath(path)[0]
-
-    def _get_docket_numbers(self):
-        for html_tree in self.html:
-            page_records_count = self._get_opinion_count(html_tree)
-            for record in range(page_records_count):
-                path = "id('ctl00_ContentPlaceHolder1_grdDocuments_ctl00__{n}')/td[5]//text()[contains(., '-')]".format(
-                    n=record
-                )
-                yield html_tree.xpath(path)[0]
-
-    @staticmethod
-    def _get_opinion_count(html_tree):
-        return int(
-            html_tree.xpath(
-                "count(id('ctl00_ContentPlaceHolder1_grdDocuments_ctl00')"
-                "//tr[contains(., 'Opinion') or contains(., 'Order')])"
+                # In texas we also have to ping the case page to get the name
+                # this is unfortunately part of the process.
+                self.url = row.xpath(".//a")[2].get("href")
+                self.method = "GET"
+                self.parameters = {}
+                self.html = self._download()
```

Comment:
This should be done in a DeferringList, I think.

Reply:
grr deferring list.

Reply:
I dont think a deferring list is appropriate here.  But perhaps im wrong.  In this case the second download is required to complete the scrape.  we need it to get the name.  I assume deferring list is more about whether we should paginate to the next page for opinions?

Reply:
This code goes to a page per case. If that's what's happening, usually we want to use a DeferringList for that.